part_02/DictBinTree.java

```diff
@@ -25,7 +25,7 @@ public boolean search(int k) {
     }
 
     private boolean search(Node x, int k) { 
-        if (x.key == k) { 
+        if (x == null || x.key == k) { 
```

Comment:
By doing this all cases will be true, even if the number isn't present.
```suggestion
        if (x == null) {
            return false;
        }
        if (k == x.key) {
            return true;
        }
```

Reply:
The pseudo-code can't be trusted